### PR TITLE
docs(features): update /features/fs page

### PR DESCRIPTION
**Pull request by pewsheen**
#### What kind of changes does this PR include?

- New or updated content
  - Update `/features/fs` content

#### Description

- What does this PR change? Give us a brief description.
  - Update `/features/fs` content

I started thinking listing the permission is a bad idea 🤣 
Maybe we could paste a link to a auto-generated permission list in `plugins-workspace`, like https://github.com/tauri-apps/plugins-workspace/blob/v2/plugins/fs/permissions/autogenerated/reference.md

### CODE REVIEW

**Comment by dklassic** [comment]
Haha that list sure is painfully long to go through

File: src/content/docs/features/file-system.mdx [hunk 1]
```diff
@@ -3,11 +3,498 @@ title: File System
...
+       .expect("error while running tauri application");
+}
+```
+
+  </TabItem>
+</Tabs>
+
+## Security
+
+This module prevents path traversal, not allowing absolute paths or parent dir components (i.e. `/usr/path/to/file` or `../path/to/file` paths are not allowed). Paths accessed with this API must be relative to one of the base directories so if you need access to arbitrary filesystem paths, you must write such logic on the core layer instead.
```
Comment (dklassic) [inline src/content/docs/features/file-system.mdx]:
```suggestion
This module prevents path traversal, not allowing absolute paths or parent dir components (i.e. `/usr/path/to/file` or `../path/to/file` paths are not allowed). Paths accessed with this API must be relative to one of the base directories so if you need access to arbitrary file system paths, you must write such logic on the core layer instead.
```

For consistency's sake let's use `file system` instead.

File: src/content/docs/features/file-system.mdx [hunk 2]
```diff
@@ -3,11 +3,497 @@ title: File System
...
+      "allow": [{ "path": "$APPDATA/*" }]
+    }
+	]
+}
+```
+
+### Default Permissions
+
+This configuration file defines the default permissions granted
+to the filesystem.
```
Comment (dklassic) [inline src/content/docs/features/file-system.mdx]:
```suggestion
This configuration file defines the default permissions granted to the filesystem.
```
For consistency's sake let's use `file system` instead.

File: src/content/docs/features/file-system.mdx [hunk 3]
```diff
@@ -3,11 +3,497 @@ title: File System
...
+
+#### Granted Permissions
+
+This default permission set enables all read-related commands and
+allows access to the `$APP` folder and sub directories created in it.
+The location of the `$APP` folder depends on the operating system,
+where the application is run.
```
Comment (dklassic) [inline src/content/docs/features/file-system.mdx]:
```suggestion
This default permission set enables all read-related commands and allows access to the `$APP` folder and sub directories created in it. The location of the `$APP` folder depends on the operating system, where the application is run.
```
Some weird multiline formatting here. Is there's reason for this?

File: src/content/docs/features/file-system.mdx [hunk 4]
```diff
@@ -3,11 +3,497 @@ title: File System
...
+
+In general the `$APP` folder needs to be manually created
+by the application at runtime, before accessing files or folders
+in it is possible.
```
Comment (dklassic) [inline src/content/docs/features/file-system.mdx]:
```suggestion
In general the `$APP` folder needs to be manually created by the application at runtime, before accessing files or folders in it is possible.
```
Some weird multiline formatting here. Is there's reason for this?

File: src/content/docs/features/file-system.mdx [hunk 5]
```diff
@@ -3,11 +3,497 @@ title: File System
...
+
+#### Denied Permissions
+
+This default permission set prevents access to critical components
+of the Tauri application by default.
+On Windows the webview data folder access is denied.
```
Comment (dklassic) [inline src/content/docs/features/file-system.mdx]:
```suggestion
This default permission set prevents access to critical components of the Tauri application by default. On Windows the webview data folder access is denied.
```
Some weird multiline formatting here. Is there's reason for this?

File: src/content/docs/features/file-system.mdx [hunk 6]
```diff
@@ -3,11 +3,497 @@ title: File System
...
+## Security
+
+This module prevents path traversal, not allowing absolute paths or parent dir components (i.e. `/usr/path/to/file` or `../path/to/file` paths are not allowed). Paths accessed with this API must be relative to one of the base directories so if you need access to arbitrary filesystem paths, you must write such logic on the core layer instead.
+
+See [@tauri-apps/plugin-fs - Security](/references/v2/js/fs/#security) for more information.
+
+## Permissions
+
+By default all plugin commands are blocked and cannot be accessed.
+You must define a list of permissions in your `capabilities` configuration.
```
Comment (dklassic) [inline src/content/docs/features/file-system.mdx]:
```suggestion
By default all plugin commands are blocked and cannot be accessed. You must define a list of permissions in your `capabilities` configuration.
```
Some weird multiline formatting here. Is there's reason for this?

**Review by dklassic — CHANGES_REQUESTED** [review]
LGTM, except some formatting and consistency nitpicking around.

File: src/content/docs/features/file-system.mdx (hunk 5 above)
Comment (pewsheen) [reply src/content/docs/features/file-system.mdx]: I just copy-paste from another place

**Review by dklassic — APPROVED** [review]

**Comment by dklassic** [comment]
So CI auto formatting seems to break the permission list in production, salvaging now.

Should be fixed in:
- #1889

**Comment by dklassic** [comment]
Closed #1478

**Pull request merged**